File: src/main/java/com/systemservices/kata/repository/EmployeeRepository.java
package com.systemservices.kata.repository;

import com.systemservices.kata.model.Employee;
import java.util.List;
import java.util.Optional;

public interface EmployeeRepository {

  List<Employee> findAll();

  Optional<Employee> findById(Long id);

  Employee save(Employee employee);

  void deleteById(Long id);

  boolean existsById(Long id);
}

File: openspec-py/app/repository/employee_repository.py
from abc import ABC, abstractmethod
from typing import Optional

from app.model.employee import Employee


class EmployeeRepository(ABC):
    @abstractmethod
    def find_all(self) -> list[Employee]: ...

    @abstractmethod
    def find_by_id(self, id: int) -> Optional[Employee]: ...

    @abstractmethod
    def save(self, employee: Employee) -> Employee: ...

    @abstractmethod
    def delete_by_id(self, id: int) -> None: ...

    @abstractmethod
    def exists_by_id(self, id: int) -> bool: ...

